### ISSUE TITLE
Rename NPM package name and GitHub repo name

### ISSUE THREAD

**Issue by nop33** [alephium/js-sdk#70]
Inspired by web3 ([npm](https://www.npmjs.com/package/web3), [GitHub](https://github.com/ChainSafe/web3.js), [docs](https://web3js.readthedocs.io/)), we want to rename alephium-js.
1. **Rename npm package to `alephium`.** If in the future we decide to split the SDK into multiple packages (like web3 does), then each of them could be named `alephium-[subpackageName]` and the main one would remain `alephium`. Important to note is that to rename an npm package, we need to deprecate the current one and publish a new one ([source](https://stackoverflow.com/questions/28371669/renaming-a-published-npm-module)).
2. **Rename the GitHub project to `alephium.js`**

**Comment by nop33** [comment]
**TL;DR: I guess we won’t go for alephium.js anymore, cause @alephium/alephium.js seems a bit repetitive. @alephium/sdk would be an alternative.**

---

@lf94 raised the topic of NPM orgs for scoping:
> I still havent seen anyone mention using an npm org, so we can do something like @alephium/sdk

To which I did some research:

> Projects that don’t use NPM orgs for scoping:
> - [near](https://www.npmjs.com/package/near-api-js) (they go for near- prefix some of their packages ([1](https://www.npmjs.com/package/near-api-js), [2](https://www.npmjs.com/package/near-cli), [3](https://www.npmjs.com/package/near-sdk-as))
> - [web3](https://github.com/ChainSafe/web3.js/tree/1.x/packages) (they chose to use the prefix of web3- for each of their packages)
> 
> Projects that use NPM orgs for scoping:
> - [near-eth](https://www.npmjs.com/search?q=%40near-eth)
> - [ethereum](https://www.npmjs.com/search?q=%40ethereumjs)
> - [ergo](https://www.npmjs.com/search?q=%40ergoplatform)
> - [solana](https://www.npmjs.com/search?q=%40solana)
> - [avalance](https://www.npmjs.com/search?q=%40avalabs) (even thought they don’t for their [main SDK lib](https://www.npmjs.com/package/avalanche))
> - [polkadot](https://www.npmjs.com/search?q=%40polkadot)
>
> Are there any other projects we should keep an eye on for inspiration?
Based on the above research, and for the argument of increased security that @lf94 mentioned, I am also in favour of an @alephium NPM org for scoping our NPM packages.
Moreover, I totally agree that since we’ll be renaming the alephium-js package, I would start from 0.0.x.
Lastly, if we are to use @alephium/[project-name], I guess we won’t go for alephium.js anymore, cause @alephium/alephium.js seems a bit repetitive. @alephium/sdk would be an alternative, as @lf94 suggested. We could later on split it into smaller packages, like @alephium/utils, @alephium/wallet-sdk, etc.

**Comment by lf94** [comment]
For the package name I vote for `@alephium/sdk` and for repo name either we use a mono repo for all SDK support (easier to derive things like OpenAPI-derived code for multiple languages at once) called `sdk` or alternatively just `js-sdk`.

**Comment by nop33** [comment]
I propose the following. It's the most consistent and most representative imo.
- `@alephium/sdk` for NPM package name
- `alephium-sdk` for GitHub project name

Other ideas that have been heard are:
- `sdk` for GitHub project name and `@alephium/sdk` for NPM)
- `js-sdk` for GitHub project name (`@alephium/sdk` for NPM)
- `alephium.js` for GitHub project name (`@alephium/alephium.js` for NPM)

**Comment by nop33** [comment]
Once the `@alephium/sdk` package is published on NPM (PR for rename is open #86), we need to deprecate the `alephium-js` one and redirect users to install the new package. We can do that with the [deprecate command](https://docs.npmjs.com/deprecating-and-undeprecating-packages-or-package-versions#using-the-command-line):

The owner of the package (is that @polarker?) needs to run the following command:

```
npm deprecate alephium-js "WARNING: This project has been renamed to @alephium/sdk. Install using @alephium/sdk instead."
```

**Comment by polarker** [comment]
@nop33 ping me when it's ready to deprecate

**Comment by nop33** [comment]
@polarker I need your feedback here so that I can proceed with the rename and then deprecation: 
- https://github.com/alephium/js-sdk/pull/86
- https://github.com/alephium/js-sdk/pull/87

**Comment by mvaivre** [comment]
To be clear, we have decided to go with:

- `@alephium/sdk` for the NPM package name
- `js-sdk` for the GitHub project name

@polarker little ping! We'd like to get your validation before proceeding. Thanks! 🙏

**Comment by nop33** [comment]
alephium-js has been deprecated, new package is live, github repo is renamed.

Closing issue.
